packages/app/src/pages/CreateReturn.tsx

```diff
@@ -0,0 +1,173 @@
+import { FormReturn } from '#components/FormReturn'
+import { ScrollToTop } from '#components/ScrollToTop'
+import { appRoutes } from '#data/routes'
+// import { useCreateReturn } from '#hooks/useCreateReturn'
+import { useCreateReturnLineItems } from '#hooks/useCreateReturnLineItems'
+import { useOrderDetails } from '#hooks/useOrderDetails'
+import { useReturnableList } from '#hooks/useReturnableList'
+import {
+  Button,
+  EmptyState,
+  PageLayout,
+  ResourceAddress,
+  Section,
+  SkeletonTemplate,
+  Spacer,
+  Stack,
+  useCoreApi,
+  useCoreSdkProvider,
+  useTokenProvider
+} from '@commercelayer/app-elements'
+import { type Address, type Return } from '@commercelayer/sdk'
+import { useEffect, useState } from 'react'
+import { Link, useLocation, useRoute } from 'wouter'
+
+export function CreateReturn(): JSX.Element {
+  const { canUser } = useTokenProvider()
+  const [, setLocation] = useLocation()
+  const [, params] = useRoute<{ orderId: string }>(appRoutes.return.path)
+
+  const [returnObj, setReturnObj] = useState<Return>()
+  const [returnNeedsCreation, setReturnNeedsCreation] = useState(false)
+
+  const orderId = params?.orderId
+  const goBackUrl =
+    orderId != null
+      ? appRoutes.details.makePath(orderId)
+      : appRoutes.home.makePath()
+
+  const { order, isLoading, mutateOrder } = useOrderDetails(orderId ?? '')
+
+  const { sdkClient } = useCoreSdkProvider()
+  const { data: createdReturnObj } = useCoreApi(
+    'returns',
+    'create',
+    returnNeedsCreation
+      ? [
+          {
+            order: sdkClient.orders.relationship(order.id)
+          },
+          {
+            include: ['origin_address', 'destination_address']
+          }
+        ]
+      : null
+  )
+
+  useEffect(() => {
+    if (!isLoading) {
+      const draftReturnObject = order.returns?.filter(
+        (returnObj) =>
+          returnObj.status === 'draft' &&
+          (returnObj.return_line_items ?? []).length === 0
+      )[0]
+      if (draftReturnObject != null) {
+        setReturnObj(draftReturnObject)
+      } else {
+        setReturnNeedsCreation(true)
+      }
+    }
+  }, [order, isLoading])
+
+  useEffect(() => {
+    if (createdReturnObj != null) {
+      setReturnObj(createdReturnObj)
+    }
+  }, [createdReturnObj])
+
+  const returnableLineItems = useReturnableList(order)
+  const {
+    createReturnLineItemsError,
+    createReturnLineItems,
+    isCreatingReturnLineItems
+  } = useCreateReturnLineItems()
+
+  if (
+    returnObj == null ||
+    order.fulfillment_status !== 'fulfilled' ||
+    returnableLineItems.length === 0 ||
```

Comment:
Actually, I am not sure what happens here if the order `isLoading`.
The mock (`makeOrder()`) has `fulfillment_status: 'unfulfilled'`
Do we need to update the mock to avoid hitting this during the fetching of real order?

Also `useReturnableList` seems to return `[]` when mocked order is passed.